en/concept/control_allocation.md

```diff
@@ -0,0 +1,70 @@
+# Control Allocation (Mixing)
+
+The PX4 architecture ensures that the airframe layout does not require special case handling in the core controllers.
+
+Mixing means to take force commands (e.g. `turn right`) and translate them to actuator commands which control motors or servos.
+For a plane with one servo per aileron this means to command one of them high and the other low.
+The same applies for multicopters: pitching forward requires changing the speed of all motors.
+
+Separating the mixer logic from the actual attitude/rate controller greatly improves reusability.
+
+<!-- https://docs.google.com/drawings/d/1Li9YhTLc3yX6mGX0iSOfItHXvaUhevO2DRZwuxPQ1PI/edit -->
+![Mixing Overview](../../assets/diagrams/mixing_overview.png)
+
+## Actuator Control Pipeline
+
+Overview of the mixing pipeline in terms of modules and uORB topics (press to show full-screen):
+<!-- https://drive.google.com/file/d/1L2IoxsyB4GAWE-s82R_x42mVXW_IDlHP/view?usp=sharing -->
+![Pipeline Overview](../../assets/concepts/control_allocation_pipeline.png)
+
+Notes:
+- The rate controller outputs torque and thrust setpoints
+- the `control_allocator` module:
+  - handles different geometries based on configuration parameters
+  - does the mixing
+  - handles motor failures
+  - publishes the motor and servo control signals
+  - publishes the servo trims separately so they can be added to the actuator testing
+- the output drivers:
+  - handle the hardware initialization and update
+  - use a shared library [src/libs/mixer_module](https://github.com/PX4/PX4-Autopilot/blob/main/src/lib/mixer_module/).
+    The driver defines a parameter prefix, e.g. `PWM_MAIN` that the library then uses for configuration.
+    Its main task is to select from the input topics and assign the right data to the outputs based on the user set `<param_prefix>_FUNCx` parameter values.
+    For example if `PWM_MAIN_FUNC3` is set to **Motor 2**, the 3rd output is set to the 2nd motor from `actuator_motors`.
+  - output functions are defined under [src/lib/mixer_module/output_functions.yaml](https://github.com/PX4/PX4-Autopilot/blob/main/src/lib/mixer_module/output_functions.yaml).
+- if you want to control an output from MAVLink, set the relevant output function to **Offboard Actuator Set x**, and then send the [MAV_CMD_DO_SET_ACTUATOR](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_ACTUATOR) MAVLink command.
+
+## Adding a new Geometry or Output Function
```

Comment:
@bkueng Assuming you are happy with MY changes I think we should merge this and iterate.

The specific part which needs more work (IMO) is this section - as I'm sure you are aware.

I am not sure this is sufficient for someone else to understand what they do for their new frame from this - say a "chinook". So I'd pull out some of the things from that doc which show the "have to do" vs the "specific case".

I really don't know what bits are most relevant, but perhaps an explanation of what you need in the YAML - e.g. https://github.com/PX4/PX4-Autopilot/commit/5cdb6fbd8e1352dcb94bd58918da405f8ff930d7#diff-c89677865693eb56c0125d5b99b9cac8b7c8bf774f51260fac6c4ce22f7a2137R382

Or we can get someone else to upgrade one of the frames and ask them to work out what they think is relevant?

Reply:
Yes there's a lot that could be said, but I'm not sure what people actually need. We can also see when someone goes through this the first time.
Adding output functions was already done by several people successfully.

This probably helps as a reference for the config: https://github.com/PX4/PX4-Autopilot/blob/main/validation/module_schema.yaml#L440=

Reply:
Fair enough. I can't judge. I added https://github.com/PX4/PX4-user_guide/pull/1923/files#r914562653 to include this reference - if you're happy with that then lets merge.